src/actions/l10n.js

```diff
@@ -0,0 +1,57 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// @flow
+import type { Action, ThunkAction, Localization } from 'firefox-profiler/types';
+import {
+  fetchMessages,
+  lazilyParsedBundles,
+} from 'firefox-profiler/app-logic/l10n-bundling';
+import { negotiateLanguages } from '@fluent/langneg';
+import { ReactLocalization } from '@fluent/react';
+/**
+ * Notify that translations for the UI are being fetched.
+ */
+export function requestL10n(): Action {
+  return {
+    type: 'REQUEST_L10N',
+  };
+}
+
+/**
+ * Receive translations for a locale.
+ */
+export function receiveL10n(localization: Localization): Action {
+  return {
+    type: 'RECEIVE_L10N',
+    localization: localization,
+  };
+}
+
+export const AVAILABLE_LOCALES: Array<string> = ['en-US', 'fr'];
+
+/**
+ * This function is called when the AppLocalizationProvider is mounted.
+ * It takes the locales available and generates l10n bundles for those locales.
+ * Initially it dispatches the info that translations are now fetching and
+ * later it dispacthes the translations and updates the state
+ */
+export function setupLocalization(
+  locales: Array<string>
+): ThunkAction<Promise<void>> {
+  return async dispatch => {
+    dispatch(requestL10n());
+
+    // Setting defaultLocale to `en-US` means that it will always be the
+    // last fallback locale, thus making sure the UI is always working.
+    const languages = negotiateLanguages(locales, AVAILABLE_LOCALES, {
+      defaultLocale: 'en-US',
```

Comment:
I think ti would be good to assign this 'en-US' string to a constant string, so we can use that constant from other places too. It would be good to add this constant inside the file on app-logic too.